src/d3a/d3a_core/redis_connections/aggregator_connection.py

```diff
@@ -81,42 +81,64 @@ def aggregator_callback(self, payload):
             self.device_aggregator_mapping[message['device_uuid']] = message['aggregator_uuid']
-            success_response_message = {
+            response_message = {
                 "status": "SELECTED", "aggregator_uuid": message['aggregator_uuid'],
                 "device_uuid": message['device_uuid'],
                 "transaction_id": message['transaction_id']}
-            self.redis_db.publish(
-                "aggregator_response", json.dumps(success_response_message)
-            )
         elif message['device_uuid'] in self.device_aggregator_mapping:
             msg = f"Device already have selected " \
                   f"{self.device_aggregator_mapping[message['device_uuid']]}"
-            error_response_message = {
+            response_message = {
                 "status": "error", "aggregator_uuid": message['aggregator_uuid'],
                 "device_uuid": message['device_uuid'],
                 "transaction_id": message['transaction_id'],
                 "msg": msg
             }
-            self.redis_db.publish(
-                "aggregator_response", json.dumps(error_response_message)
-            )
         else:
             self.aggregator_device_mapping[message['aggregator_uuid']].\
                 append(message['device_uuid'])
             self.device_aggregator_mapping[message['device_uuid']] = message['aggregator_uuid']
-            success_response_message = {
+            response_message = {
                 "status": "SELECTED", "aggregator_uuid": message['aggregator_uuid'],
                 "device_uuid": message['device_uuid'],
                 "transaction_id": message['transaction_id']}
+        self.redis_db.publish(
+            "aggregator_response", json.dumps(response_message)
+        )
+
+    def _unselect_aggregator(self, message):
+        if message['device_uuid'] in self.device_aggregator_mapping and \
+                message['aggregator_uuid'] in self.aggregator_device_mapping:
+            try:
+                with self.lock:
+                    del self.device_aggregator_mapping[message['device_uuid']]
+                    self.aggregator_device_mapping[message['aggregator_uuid']]\
+                        .remove(message['device_uuid'])
+                response_message = {
+                    "status": "UNSELECTED", "aggregator_uuid": message['aggregator_uuid'],
+                    "device_uuid": message['device_uuid'],
+                    "transaction_id": message['transaction_id']}
+                self.redis_db.publish(
+                    "aggregator_response", json.dumps(response_message)
+                )
+            except Exception:
+                response_message = {
+                    "status": "error", "aggregator_uuid": message['aggregator_uuid'],
+                    "device_uuid": message['device_uuid'],
+                    "transaction_id": message['transaction_id'],
+                    "msg": "Error unselecting aggregator"
```

Comment:
For debugging, it would be good to have at lease the kind of exception in the `msg`, if not the traceback.